peachjam/views/documents.py

```diff
@@ -126,10 +126,35 @@ def render_to_response(self, context, **response_kwargs):
                         pdf,
                         "application/pdf",
                         source_file.filename_for_download(".pdf"),
+                        disposition="inline",
                     )
         raise Http404()
 
 
+class DocumentPublicationView(DocumentSourceView):
+    def render_to_response(self, context, **response_kwargs):
+        if hasattr(self.object, "publication_file"):
+            publication_file = self.object.publication_file
+            if publication_file.use_source_file:
+                # TODO: change document_source to display inline?
```

Comment:
See https://github.com/laws-africa/peachjam/commit/3df4b19079e4e10d7b1c1c98570cf2c7df30487c

using `inline` means the browser disregards the filenames. I think we want attachment in all cases.

Reply:
for PDF files, it works like this:

* if `content-disposition` is `inline`, then the browser will download the PDF and try to show it in the browser. If the user then clicks the "save" button in the PDF viewer, the browser chooses a filename based on the URL and ignores the `filename` part of `content-disposition` -- it sometimes seems to re-download the file
* if the `content-disposition` is `attachment`, then the browser will download the file and save it using the `filename` part of the header. The user must then open the file in their downloads to view the actual file

So for source files, because we want nice filenames, we changed to `attachment` from `inline` so that the filename was actually used.